### PR TITLE
[승희] 2주차 수강기록

### PR DESCRIPTION
**Pull request by Jin409**
### 질문
- 준영속 상태가 잘 이해되지 않음
    - 이미 DB에 한번 저장됐거나 식별자가 존재하는 엔티티 -> 그렇다면 다시 객체를 조회해서 꺼내오는 em.find() 를 해도 똑같이 변경감지가 안되어야 하는 게 아닌가..?

### CODE REVIEW

**Review by ChoiYoo — APPROVED** [review]
em.find()를 했을 경우 DB에서 조회해서 1차 캐시에 저장하기 때문에 준영속 상태에서 영속 상태로 바뀌니까 변경감지가 되는거라고 전 이해했어요!!

**Comment by Jin409** [comment]
> em.find()를 했을 경우 DB에서 조회해서 1차 캐시에 저장하기 때문에 준영속 상태에서 영속 상태로 바뀌니까 변경감지가 되는거라고 전 이해했어요!!

아앗 제가 준영속 상태를 잘못 이해하고 있었나봐요..
DB 에 한번 갔다온 엔티티 = 준영속 상태 라고 이해했는데, 한번 flush 되어서 저장된 객체는 준영속객체라는거겠죠?! 저는 아예 저장된 객체 자체를 준영속 객체로 분리한다고 이해했어요..!! 
flush 된 이후에 준영속 객체가 되니, 더 이상 영속성 컨텍스트에서 관리하지 않으니 업데이트가 감지되지 않는다.. 라고 이해해도 되려나요?!

**Comment by ChoiYoo** [comment]
> > em.find()를 했을 경우 DB에서 조회해서 1차 캐시에 저장하기 때문에 준영속 상태에서 영속 상태로 바뀌니까 변경감지가 되는거라고 전 이해했어요!!
> 
> 아앗 제가 준영속 상태를 잘못 이해하고 있었나봐요.. DB 에 한번 갔다온 엔티티 = 준영속 상태 라고 이해했는데, 한번 flush 되어서 저장된 객체는 준영속객체라는거겠죠?! 저는 아예 저장된 객체 자체를 준영속 객체로 분리한다고 이해했어요..!! flush 된 이후에 준영속 객체가 되니, 더 이상 영속성 컨텍스트에서 관리하지 않으니 업데이트가 감지되지 않는다.. 라고 이해해도 되려나요?!

저는 flush되어서 저장된 객체는 아직 영속상태이고 단순히 detach나 close로 직접 하는 경우나 트랜잭션이 종료될 때 준영속 상태가 된다고 이해하고 있었는데 생각해보니 좀 헷갈리는 부분이 있네요,, 근데 우선 준영속 객체는 식별자를 갖고 있는 엔티티이니까 DB에 한번 갔다온 엔티티가 맞는 듯 해요! 찾아보니까

식별자를 기준으로 이미 한번 영속상태가 되어버린 엔티티가 있는데, 더이상 영속성 컨텍스트가 관리하지 않으면 모두 준영속 상태입니다. em.detach()를 해서 직접적으로 준영속 상태가 될 수도 있고, 수정을 위해 html form에 데이터를 노출한 이후에 다시 new로 재조립된 엔티티일 수 도 있습니다. (https://www.inflearn.com/questions/70393/book-%EA%B0%9D%EC%B2%B4%EA%B0%80-%EC%99%9C-%EC%A4%80%EC%98%81%EC%86%8D%EC%9D%B8%EA%B2%83%EC%9D%B8%EA%B0%80)

라고 하네요,,! 활용편 섹션7 변경감지와 병합 부분에서 더 자세하게 다루는거 같아요!
